app/helpers/request_info_helper.rb

```diff
@@ -23,20 +26,45 @@ def prov_grid_data(edit, vms, vm)
     {
       :headers  => prov_grid_vm_header(edit, clones, vms),
       :rows     => rows,
-      :selected => selected_item(edit).presence || none_index,
+      :selected => selected_vm(edit).presence || none_index,
       :recordId => edit[:req_id] || "new",
     }
   end
 
-  private
+  def prov_host_grid_data(edit, hosts)
+    none_index = '__HOST__NONE__'
+    rows = [{:id => none_index, :clickable => true, :cells => none_cells(5)}]
+    rows += hosts.map do |h|
+      {:id => h.id.to_s, :clickable => true, :cells => prov_host_grid_cells(h, edit)}
+    end
+    {
+      :headers  => prov_grid_host_header(edit),
+      :rows     => rows,
+      :selected => selected_host(edit).presence || none_index,
+      :recordId => (edit && edit[:req_id]) || "new",
+    }
+  end
 
   def prov_grid_vm_header(edit, clones, vms)
     headers = []
     edit[:vm_columns].each_with_index do |h, index|
       item = {:text => edit[:vm_headers][h], :header_text => edit[:vm_headers][h]}
       if vms && clones.exclude?(edit[:wf].request_type)
         item[:sort_choice] = h
-        item[:sort_data] = sort_data(edit, index)
+        item[:sort_data] = sort_data(edit, index, 'vm')
+      end
+      headers.push(item)
+    end
+    headers
+  end
+
+  def prov_grid_host_header(edit)
```

Comment:
This method can probably be merged into the prov_grid_vm_header method and take a `type` parameter, but that's super-minor.

Reply:
Yes, there are few more similar tables than follows a similar structure. Will try to make them into a common method when the others are also done...